src/Mod/Sketcher/Gui/CommandConstraints.cpp

```diff
@@ -864,64 +864,109 @@ class DrawSketchHandlerGenConstraint: public DrawSketchHandler
+            newSelType = SelRoot;
+        }
+        else if (allowedSelTypes & SketcherGui::SelVertex && VtId != -1) {
+            sketchgui->getSketchObject()->getGeoVertexIndex(VtId,
+                                                            selIdPair.GeoId,
+                                                            selIdPair.PosId);
+            newSelType = SelVertex;
+        }
+        else if (allowedSelTypes & SketcherGui::SelEdge && CrvId != -1) {
+            selIdPair.GeoId = CrvId;
+            newSelType = SelEdge;
+        }
+        else if (allowedSelTypes & SketcherGui::SelHAxis && CrsId == 1) {
+            selIdPair.GeoId = Sketcher::GeoEnum::HAxis;
+            newSelType = SelHAxis;
+        }
+        else if (allowedSelTypes & SketcherGui::SelVAxis && CrsId == 2) {
+            selIdPair.GeoId = Sketcher::GeoEnum::VAxis;
+            newSelType = SelVAxis;
+        }
+        else if (allowedSelTypes & SketcherGui::SelExternalEdge) {
+            //TODO: Figure out how this works
+            newSelType = SelExternalEdge;
+        }
+
+        if (selIdPair.GeoId == Constraint::GeoUndef) {
+            // If mouse is released on "blank" space, start over
+            selSeq.clear();
+            resetOngoingSequences();
+            Gui::Selection().clearSelection();
+        }
+        else {
+            // TODO: If mouse is released on something allowed, select it and move forward
+            selSeq.push_back(selIdPair);
+
+            _tempOnSequences->clear();
+            allowedSelTypes = 0;
+            for (std::set<int>::iterator token = ongoingSequences->begin();
+                 token != ongoingSequences->end(); ++token) {
+                if ((cmd->allowedSelSequences).at(*token).at(seqIndex) == newSelType) {
+                    if (seqIndex == (cmd->allowedSelSequences).at(*token).size()-1) {
+                        // TODO: One of the sequences is completed. Pass to cmd->applyConstraint
+                        cmd->applyConstraint(selSeq, *token); // TODO: replace arg 2 by ongoingToken
+
+                        selSeq.clear();
+                        resetOngoingSequences();
+
+                        return false;
+                    }
+                    _tempOnSequences->insert(*token);
+                    allowedSelTypes = allowedSelTypes | (cmd->allowedSelSequences).at(*token).at(seqIndex+1);
+                }
+            }
+
+            // TODO: Progress to next seqIndex
+            std::swap(_tempOnSequences, ongoingSequences);
+            seqIndex++;
+        }
+
         return false;
     }
 
 protected:
+    GenericConstraintSelection* selFilterGate;
+
     const char** constraintCursor;
     CmdSketcherConstraint* cmd;
 
+    std::vector<SelIdPair> selSeq;
+    int allowedSelTypes = 0;
+
     /// indices of currently ongoing sequences in cmd->allowedSequences
-    std::set<int> ongoingTokens;
+    std::set<int> *ongoingSequences, *_tempOnSequences;
```

Comment:
I just kept it that way since the code felt cleaner that way. For eg, I wasn't able to use "new std::set<int>()" and could pass them directly to swap. I'm sorry I don't really know the difference between creating on heap and creating on stack.